File: src/list/moveZerosToTheEnd.ts
export default function moveZerosToTheEnd(list: number[]) {
  for (let index = 0; index < list.length; index++) {
    const external = list[index];
    if (external !== 0) continue;
    for (let i = index + 1; i < list.length; i++) {
      const internal = list[i];
      if (internal === 0) continue;
      list[i] = external;
      list[index] = internal;
      break;
    }
  }
  return list;
}

File: src/list/move_zeros_to_the_end.py
def move_zeros_to_the_end(input: list[int]):
  for index, external in enumerate(input):
    if external != 0: continue
    for i in range(index + 1, len(input)):
      internal = input[i]
      if internal == 0: continue
      input[i] = 0
      input[index] = internal
      break
  return input


